### DISCUSSION TITLE
Is there a Pytest equivalent to the unittest unittest.TestLoader.loadTestsFromModule() method?

### DISCUSSION THREAD

**Discussion by SchoofsEbert** [pytest-dev/pytest#9138] [Q&A] [answered]
I am working on a program that modifies test suites.  
In the original implementation, there was only support for the `unittest` framework. I am trying to add support for Pytest now.  
  
With the default `unitest` module, I could keep the modified tests as ASTs in memory, which I convert to a module, and finally, I can create a test suite from that, and run it.
```python
#create module
code = test.AST.compile()
module = types.ModuleType(test.name)
module.__dict__.update(scope)
exec(code, module.__dict__)
#create suite
suite = unittest.TestLoader().loadTestsFromModule(module)
#run test
suite.run()
```    
From everything that I can find in the Pytest documentation, this seems not possible.  
Pytest needs the testfiles to function.  
  
Does someone know if this is either possible for Pytest or a possible workaround? I don't want to write hundredths of modified tests to the filesystem, as this would be more expensive than keeping them in memory.

**Comment by nicoddemus** [comment, marked as answer]
One way I can think of now is to implement your own `pytest_pycollect_makemodule` hook, which would return a custom `Module` object. Here's the default implementation that can be used as basis:

https://github.com/pytest-dev/pytest/blob/7720154ca023da23581d87244a31acf5b14979f2/src/_pytest/python.py#L210-L215

**Reply by SchoofsEbert** [reply]
Thank you, this is indeed the direction to go. It will still be a lot of work to add support for multiple modules, and especially in the case of multiprocessing, but this is the way to go.
